src/Compilers/Core/Portable/FileSystem/PathUtilities.cs

```diff
@@ -436,6 +436,34 @@ public static string CombinePathsUnchecked(string root, string? relativePath)
             return root + relativePath;
         }
 
+        /// <summary>
+        /// Combines paths with the same semantics as <see cref="Path.Combine(string, string)"/>
+        /// but does not throw on null paths or paths with invalid characters.
+        /// </summary>
+        /// <param name="root">First path: absolute, relative, or null.</param>
+        /// <param name="path">Second path: absolute, relative, or null.</param>
+        /// <returns>
+        /// The combined paths. If <paramref name="path"/> contains an absolute path, returns <paramref name="root"/>.
```

Comment:
Doesn't it return `path` when it's an absolute path?

Reply:
Yes.